plugins/obs-outputs/librtmp/rtmp.c

```diff
@@ -928,11 +929,19 @@ RTMP_Connect0(RTMP *r, struct sockaddr * service, socklen_t addrlen)
 
     /* set timeout */
     {
-        SET_RCVTIMEO(tv, r->Link.timeout);
+        SET_RCVTIMEO(tvr, r->Link.receiveTimeout);
         if (setsockopt
-                (r->m_sb.sb_socket, SOL_SOCKET, SO_RCVTIMEO, (char *)&tv, sizeof(tv)))
+                (r->m_sb.sb_socket, SOL_SOCKET, SO_RCVTIMEO, (char *)&tvr, sizeof(tvr)))
         {
-            RTMP_Log(RTMP_LOGERROR, "%s, Setting socket timeout to %ds failed!",
+            RTMP_Log(RTMP_LOGERROR, "%s, Setting socket receive timeout to %ds failed!",
+                     __FUNCTION__, r->Link.timeout);
+        }
+
+        SET_SNDTIMEO(tvs, r->Link.sendTimeout);
+        if (setsockopt
+                (r->m_sb.sb_socket, SOL_SOCKET, SO_SNDTIMEO, (char *)&tvs, sizeof(tvs)))
+        {
+                RTMP_Log(RTMP_LOGERROR, "%s, Setting socket send timeout to %ds failed!",
                      __FUNCTION__, r->Link.timeout);
```

Comment:
The `RTMP_Log` calls need to be changed to use the new timeout fields, this won't compile as-is.

Reply:
On my system (Ubuntu) the actual timeout is double whatever the configured timeout is. It seems to be a common problem which is why I set the number. Just increased it to 6 seconds since Windows probably behaves normally.

Reply:
Fixed